### ISSUE TITLE
Implement support for PDB+(MOL/SMI) in RDKitToolKitWrapper.from_file()

### ISSUE THREAD

**Issue by j-wags** [openforcefield/openforcefield#121]
Currently I've made Molecule.from_file() raise an exception if the user passes in a PDB file and asks for it to be read by a RDKitToolkitWrapper. This is because RDKit can not perceive bond orders/aromaticity from PDB CONECT records or geometry (which is dangerous to begin with). 

However, it seems to be possible for RDKit to load coordinates from a PDB, and then get bond information from a MOL or SMI file of the same molecule. This could be a useful input method. 

More information: https://github.com/rdkit/rdkit/issues/1031#issuecomment-240308113

**Comment by j-wags** [comment]
Example of implementation https://github.com/ADicksonLab/mastic/blob/58749c40fe364110e3e7be8aa79a89f32d956d09/mastic/interfaces/rdkit.py#L333

**Comment by hjuinj** [comment]
yeah I think the `AssignBondOrdersFromTemplate` function is really useful, I use it quite often with SMILES input as reference mol. My understanding is there is not something similar in openeye?

**Comment by j-wags** [comment]
I think you're correct that openeye doesn't have a matching function, though that might be worth looking into. Right now my plan for OE is to let it load PDB files directly and try to perceive bond orders from geometry (which is risky, maybe we should warn users as well). 

(OE bond order perception: https://docs.eyesopen.com/toolkits/python/oechemtk/connectivity.html )

**Comment by davidlmobley** [comment]
@j-wags we should talk about perceiving bond orders from a PDB based on geometry in person; there can be major issues.

Probably we should e-mail OpenEye support and ask if there's an equivalent to that RDKit thing, or if they have a good workaround. I bet they do. 

(Offhand, I would actually think you could do it by loading the SMILES into an `OEMol`, then matching that onto the chemical graph from the PDB and then pulling in the coordinates from the PDB into your `OEMol`.)

Actually, I think the current process of matching template molecules to the topology in the current master branch of `openforcefield` basically does what you need (it's a graph matching algorithm).

**Comment by j-wags** [comment]
Thanks for the lead, @davidlmobley  -- posting this here for my own reference: https://github.com/openforcefield/openforcefield/blob/master/openforcefield/typing/engines/smirnoff/forcefield.py#L329

I'll poke around a bit to see if OE has an equivalent to the RDKit thing, and email OE if not.

Yes -- I think letting folks make molecules from a PDB file alone is risky, and worth discussion (broad usability vs. potential mistakes). Let's chat about this tomorrow in Irvine.

**Comment by davidlmobley** [comment]
We forgot to talk about the general "make molecules from a PDB" philosophical issues. Let's try to discuss next week sometime, maybe a call Wednesday or in person Friday.

**Comment by hjuinj** [comment]
@j-wags just been reading some CELPP code and they (your old colleagues?)  did something akin to `AssignBondOrdersFromTemplate` in openeye using [mcs search to find the atom mapping between two structures](https://github.com/drugdata/D3R/blob/9e305ba7c78d4979666db303cc4a190ced6afa59/d3r/evaluate.py#L164). I think for our purpose this code would work too though I have not yet tested it.

**Comment by jchodera** [comment]
We also have code for this (which does *not* attempt to always pick the same atom ordering if multiple equivalent matches are possible) in the [existing toolkit branch](https://github.com/openforcefield/openforcefield/blob/master/openforcefield/typing/engines/smirnoff/forcefield.py#L329-L373) that uses graph isomorphism matching from the [NetworkX toolkit](https://networkx.github.io/).
